openwillis-face/src/openwillis/face/face_landmark.py

```diff
@@ -880,68 +907,61 @@ def facial_expressivity(
-            column is the overall framewisedisplacement measurement as a mean of all previous displacement columns.
+    This function uses MediaPipe Face Mesh to detect and analyze facial landmarks in a video.
+    It can optionally normalize the landmarks using a baseline video and split the analysis
+    by speaking probability.
 
-        summary : pandas.DataFrame
-            dataframe with summary measurements. first column is name of statistic, subsequent columns are all
-            facial landmarks, last column is overall column with composite measures for all landmarks. first
-            row contains sum of framewise displacement values, second row contains mean framewise displacement
-            over the video, and third row has standard deviation of framewise displacement. in case an optional
-            baseline video was provided, all summary measures are relative to baseline values calculated from
-            baseline video.
+    Parameters
+    ..........
+    filepath : str
+        Path to the input video file
+    baseline_filepath : str, optional
+        Path to a baseline video for normalization. Default is empty string.
+    bbox_list : list of dicts, optional
+        List of bounding box dictionaries for each frame in the video.
+    base_bbox_list : list of dicts, optional
+        List of bounding box dictionaries for each frame in the baseline video.
+    frames_per_second : float, optional
+        The number of frames to sample per second of video. Default is 3.
+        This determines the temporal resolution of the analysis.
+    normalize : bool, optional
+        Whether to normalize the landmarks. Default is True.
+    align : bool, optional
+        Whether to align the landmarks. Default is False.
+    rolling_std_seconds : int, optional
+        Window size in seconds for calculating rolling standard deviation of speaking probability.
+        Default is 3.
+    split_by_speaking : bool, optional
+        Whether to split the analysis by speaking probability. Default is False.
+
+    Returns
+    ..........
+    df_landmark : pandas.DataFrame
+        Dataframe containing landmark coordinates for each frame of the video
+    df_summ : pandas.DataFrame
+        Summary statistics of the landmark movements
 
     ---------------------------------------------------------------------------------------------------
     """
     config = get_config(os.path.abspath(__file__), "facial.json")
 
     try:
+        df_landmark = get_landmarks(
+            filepath,
+            bbox_list=bbox_list,
+            frames_per_second=frames_per_second
+        )
 
-        df_landmark = get_landmarks(filepath, bbox_list=bbox_list)
-        
         if normalize:
             df_landmark = normalize_face_landmarks(df_landmark, align=align)
+        
+        # only pass in frames that are not null
+        sampled_frames = df_landmark.dropna()
```

Comment:
```suggestion
        sampled_frames = df_landmark.dropna().reset_index()
```

Need to reset index after filtering out rows, otherwise get some weird behavior in predicting speaking vs not speaking frames.